Source/GoBot/ViewDatabase.swift

```diff
@@ -1328,12 +1369,47 @@ class ViewDatabase {
         
         return try self.mapQueryToKeyValue(qry: qry, useNamespacedTables: true)
     }
-    
-    func reports(limit: Int = 200) throws -> [Report] {
-        guard let db = self.openDB else {
+
+    // MARK: - Reports
+
+    /// Returns the total number of unread reports
+    func countNumberOfUnreadReports() throws -> Int {
+        guard let connection = self.openDB else {
+            throw ViewDatabaseError.notOpen
+        }
+        let queryString = """
```

Comment:
In general I think we should prefer to use SQLite.swift's semantic queries (like `reports.join(.left, read_messages).filter(...`) because that lets the compiler check that all the table names and column names are correct. What do you think?

Reply:
I don't think it checks the table and column names are correct in compile time. We should discuss this sometime, there are some things I don't like about it and there are some things we cannot do without it.

Reply:
Ok yeah let's discuss at some point in the future.